### PR TITLE
Fix listing images on DO printing raw bytes

### LINKED ISSUE #716: Listing images on DigitalOcean returns an array of bytes
## Description
When running `./ops image list -t do` it prints an array of bytes :
```bash
ahmedaabouzied@euclid:~/work/opensource/ops$ ./ops image list -t do
[123 34 105 100 34 58 32 34 117 110 97 117 116 104 111 114 105 122 101 100 34 44 32 34 109 101 115 115 97 103 101 34 58 32 34 85 110 97 98 108 101 32 116 111 32 97 117 116 104 101 110 116 105 99 97 116 101 32 121 111 117 46 34 32 125]
```
It should print the images in a table as it does with local images 
```bash
ahmedaabouzied@euclid:~/work/opensource/ops$ ./ops image list -t onprem
+----------+-------------------------------------------+---------+-------------+
|   NAME   |                   PATH                    |  SIZE   |  CREATEDAT  |
+----------+-------------------------------------------+---------+-------------+
| main.img | /home/ahmedaabouzied/.ops/images/main.img | 20.4 MB | 4 weeks ago |
+----------+-------------------------------------------+---------+-------------+
| node.img | /home/ahmedaabouzied/.ops/images/node.img | 58.7 MB | 4 weeks ago |
+----------+-------------------------------------------+---------+-------------+
| webg.img | /home/ahmedaabouzied/.ops/images/webg.img | 21.0 MB | 4 weeks ago |
+----------+-------------------------------------------+---------+-------------+
| webs.img | /home/ahmedaabouzied/.ops/images/webs.img | 12.1 MB | 4 weeks ago |
+----------+-------------------------------------------+---------+-------------+
```

This is due to printing the raw buffer here 
https://github.com/nanovms/ops/blob/20cf3962b75546180c2a8040dadf8275afe9a7f4/lepton/digital_ocean.go#L124

### PR DESCRIPTION
**Pull request by ahmedaabouzied**
## Issues
Fixes #716 

## Changes
* Fix the problem of printing raw bytes when listing images in Digital Ocean.
* Use Digital Ocean official SDK instead of calling the API directly.
  This SDK parses the responses from Digital Ocean into structs, saving
  the time wasted in parsing responses and errors from Digital Ocean
  when there is a solution for it already.
* Print the list of images in a table.
* Add a unit test using [Go examples](https://golang.org/pkg/testing/#hdr-Examples) that ensures that the printed output to stdout is
  correct. Ideally the printing logic should be separated from the listing of images, but that can be postponed for later.

Now listing Digital Ocean images prints as follows:
```bash
ahmedaabouzied@euclid:~/work/opensource/ops$ ./ops image list -t do
+--------------------------------+-----------+----------------------+
|              NAME              |  STATUS   |       CREATED        |
+--------------------------------+-----------+----------------------+
| 6.9 x32                        | available | 2018-01-30T12:26:21Z |
+--------------------------------+-----------+----------------------+
| 6.9 x64                        | available | 2018-06-02T00:21:33Z |
+--------------------------------+-----------+----------------------+
| 16.04.6 (LTS) x32              | available | 2019-10-29T00:04:01Z |
+--------------------------------+-----------+----------------------+
| 12.1 x64 ufs                   | available | 2020-02-18T22:06:39Z |
+--------------------------------+-----------+----------------------+
| v1.5.6                         | available | 2020-06-18T21:03:14Z |
+--------------------------------+-----------+----------------------+
```
And when there is an error, it will print an error in red as it should instead of also printing the JSON error response.
```bash
ahmedaabouzied@euclid:~/work/opensource/ops$ export TOKEN=""
ahmedaabouzied@euclid:~/work/opensource/ops$ ./ops image list -t do
GET https://api.digitalocean.com/v2/images: 401 Unable to authenticate you.
```

### CODE REVIEW

**Comment by ahmedaabouzied** [comment]
I'll keep this as a draft while I'm working on handling error responses from Digital Ocean and printing the list of images in a table

File: lepton/digital_ocean_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		}`)
+	})
+	do := &DigitalOcean{
+		Client: client,
+	}
+	err := do.ListImages(&Context{})
+	if err != nil {
+		println(err.Error())
+	}
+	// Output:
```
Comment (eyberg) [inline lepton/digital_ocean_test.go:56]: cruft

**Comment by eyberg** [comment]
thanks - this works, although the test doesn't actually test anything so i don't think it's worth including

**Comment by ahmedaabouzied** [comment]
@eyberg, the test actually tests the printed output to match exactly with what's written in the `// output` comments using the [Examples](https://blog.golang.org/examples) tests supported in Go.

**Comment by ahmedaabouzied** [comment]
I've now separated the logic of getting the images from actually printing the images. Also added a test to test getting the images from DO and making them `CloudImage` structs.

File: lepton/digital_ocean_test.go (hunk 1 above)
Comment (ahmedaabouzied) [reply lepton/digital_ocean_test.go:99]: Printing the error will make the test fail. Printing anything other than the exact text specified after the `// Output:` comment will make the test fail.

**Comment by eyberg** [comment]
your new test, Test_do_GetImages is fine although doesn't look like Example_do_ListImages is called? so can we remove that and we'll merge

File: lepton/digital_ocean_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	mux *http.ServeMux
+
+	ctx = context.TODO()
+
+	client *godo.Client
+
+	server *httptest.Server
+)
+
+func Test_do_GetImages(t *testing.T) {
```
Comment (eyberg) [inline lepton/digital_ocean_test.go:25]: this should prob. just be called TestDoGetImages

**Comment by eyberg** [comment]
thanks!

**Pull request merged**